tool/ci/pkg/bot/bot.go

```diff
@@ -0,0 +1,157 @@
+}
+type GithubClient struct {
+	Client *github.Client
+}
+
+// New returns a new instance of  Bot
+func New(c Config) (*Bot, error) {
+	err := c.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &Bot{
+		Environment: c.Environment,
+		GithubClient: GithubClient{
+			Client: c.Environment.Client,
+		},
+	}, nil
+}
+
+// CheckAndSetDefaults verifies configuration and sets defaults
+func (c *Config) CheckAndSetDefaults() error {
+	if c.Environment == nil {
+		return trace.BadParameter("missing parameter Environment")
+	}
+	return nil
+}
+
+// invalidateApprovals dismisses all approved reviews on a pull request.
+func (c *Bot) invalidateApprovals(ctx context.Context, msg string, reviews []review) error {
+	pr := c.Environment.PullRequest
+	for _, v := range reviews {
+		if v.status == ci.Approved {
+			_, _, err := c.Environment.Client.PullRequests.DismissReview(ctx,
+				pr.RepoOwner,
+				pr.RepoName,
+				pr.Number,
+				v.id,
+				&github.PullRequestReviewDismissalRequest{Message: &msg},
+			)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+		}
+	}
+	return nil
+}
+
+// DimissStaleWorkflowRunsForExternalContributors dismisses stale workflow runs for external contributors
+func (gc GithubClient) DimissStaleWorkflowRunsForExternalContributors(ctx context.Context, token, repoOwner, repoName string) error {
+	pulls, _, err := gc.Client.PullRequests.List(ctx, repoOwner, repoName, &github.PullRequestListOptions{State: ci.Open})
+	if err != nil {
+		return err
+	}
+	for _, pull := range pulls {
+		err := gc.DismissStaleWorkflowRuns(ctx, token, *pull.Base.User.Login, *pull.Base.Repo.Name, *pull.Head.Ref)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
+// DismissStaleWorkflowRuns dismisses stale Check workflow runs.
+// Stale workflow runs are workflow runs that were previously ran and are no longer valid
+// due to a new event triggering thus a change in state. The workflow running in the current context is the source of truth for
+// the state of checks.
+func (gc GithubClient) DismissStaleWorkflowRuns(ctx context.Context, token, owner, repoName, branch string) error {
+	var targetWorkflow *github.Workflow
+	workflows, _, err := gc.Client.Actions.ListWorkflows(ctx, owner, repoName, &github.ListOptions{})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, w := range workflows.Workflows {
+		if *w.Name == ci.CheckWorkflow {
+			targetWorkflow = w
+			break
+		}
+	}
+	list, _, err := gc.Client.Actions.ListWorkflowRunsByID(ctx, owner, repoName, *targetWorkflow.ID, &github.ListWorkflowRunsOptions{Branch: branch})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	sort.Sort(ByTime(list.WorkflowRuns))
+
+	// Deleting all runs except the most recent one.
+	for i := 0; i < len(list.WorkflowRuns)-1; i++ {
+		run := list.WorkflowRuns[i]
+		err := gc.deleteRun(ctx, token, owner, repoName, *run.ID)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
```

Comment:
These feel like they should be attached to the bot not the `GithubClient`. Then at the top level you can just do

```
err := bot.DismissWorkflows()
...
```

Reply:
@r0mant Any reason you wanted this attached to `GitHubClient`? Feels like it should be attached to `Bot`.